libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -826,6 +1011,80 @@ void AP_CRSF_Telem::process_param_write_frame(ParameterSettingsWriteFrame* write
 #endif
 }
 
+// get status text data
+void AP_CRSF_Telem::calc_status_text()
+{
+    if (!_statustext.available) {
+        WITH_SEMAPHORE(_statustext.sem);
+        // check link speed
+        if (_telem_rf_mode != AP_RCProtocol_CRSF::CRSF_RF_MODE_150HZ) {
+            // keep only warning/error/critical/alert/emergency status text messages
+            bool got_message = false;
+            while (_statustext.queue.pop(_statustext.next)) {
+                if (_statustext.next.severity <= MAV_SEVERITY_WARNING) {
+                    got_message = true;
+                    break;
+                }
+            }
+            if (!got_message) {
+                return;
+            }
+        } else {
+            if (!_statustext.queue.pop(_statustext.next)) {
+                return;
+            }
+        }
+        _statustext.available = true;
+    }
+
+    _telem_type = get_custom_telem_frame_id();
+    _telem.bcast.custom_telem.status_text.sub_type = AP_RCProtocol_CRSF::CustomTelemSubTypeID::CRSF_CUSTOM_TELEM_STATUS_TEXT;
+    _telem.bcast.custom_telem.status_text.severity = _statustext.next.severity;
+    const uint8_t len = hal.util->snprintf(_telem.bcast.custom_telem.status_text.text, 50, "%s", _statustext.next.text);
```

Comment:
ARRAY_SIZE rather than 50

Reply:
`len` here is *NOT* the number of characters written into `_telem.bcast.custom_telem.status_text.text`.

It is the number of characters `snprintf` would have written if it was able to write everything.

You may wish to re-jig all of this using `strncpy` instead...

Reply:
ok done